src/main/java/io/managed/services/test/client/registrymgmt/RegistryMgmtApiUtils.java

```diff
@@ -167,7 +216,33 @@ public static <T extends Throwable> void waitUntilRegistryIsDeleted(
         try {
             waitFor("registry instance to be deleted", ofSeconds(1), ofSeconds(20), ready);
         } catch (TimeoutException e) {
-            throw new RegistryNotDeletedException(registryAtom.get(), e);
+            throw new RuntimeException(e);
+            // throw new RegistryNotDeletedException(registryAtom.get(), e);
+        }
+    }
+
+    public static <T extends Throwable> void waitUntilRootTypeOfRegistryIsDeleted(
+            ThrowingSupplier<Optional<RootTypeForRegistry>, T> supplier)
+            throws T, InterruptedException, RegistryNotDeletedException {
+
+        var registryAtom = new AtomicReference<RootTypeForRegistry>();
+        ThrowingFunction<Boolean, Boolean, T> ready = l -> {
+            var exists = supplier.get();
+            if (exists.isEmpty()) {
+                return true;
+            }
+
+            var registry = exists.get();
+            LOGGER.debug(registry);
+            registryAtom.set(registry);
+            return false;
+        };
+
+        try {
+            waitFor("registry instance to be deleted", ofSeconds(1), ofSeconds(20), ready);
+        } catch (TimeoutException e) {
+            throw new RuntimeException(e);
```

Comment:
back to `RegistryNotDeletedException`

Reply:
I had to create a new exception `RootTypeOfRegistry` not deleted. Just wanted to know what purpose do the types `Registry`and `RootTypeOfRegistry` serve.